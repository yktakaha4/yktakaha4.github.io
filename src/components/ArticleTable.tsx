import dayjs from "dayjs";
import {FC} from "react";

export interface ArticleTableRow {
    title: string
    url: string
    likes: number
    publishedAt: Date
}

export interface ArticleTableProps {
    rows: Array<ArticleTableRow>
}

export const ArticleTable: FC<ArticleTableProps> = ({ rows }) => {
    return (
        <table>
            <thead>
            <tr>
                <th style={{ textAlign: 'center' }}>公開日</th>
                <th>記事名</th>
            </tr>
            </thead>
            <tbody>
            {
                rows.map(row => (
                    <tr>
                        <td>{dayjs(row.publishedAt).format('YYYY/M/D')}</td>
                        <td><a href={row.url} target="_blank" rel="noopener noreferrer">{row.title}</a></td>
                    </tr>
                ))
            }
            </tbody>
        </table>
    )
}
