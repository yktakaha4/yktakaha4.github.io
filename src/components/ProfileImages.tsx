import { FC } from 'react';
import { Avatar } from '@/components/ui/Avatar';

export const ProfileImages: FC = () => {
  const src = [
    //
    require('@site/static/img/icon.jpg').default,
    require('@site/static/img/yktakaha4.jpg').default,
  ];
  return <Avatar src={src} />;
};
