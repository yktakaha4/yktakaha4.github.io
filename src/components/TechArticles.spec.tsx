import { render } from '@testing-library/react';
import { TechArticles } from '@/components/TechArticles';
import techArticles from '@/components/data/techArticles.json';

jest.mock('@/components/data/techArticles.json', () => {
  const itemsCount = 30;
  const baseDate = new Date('2020-01-01');
  const publishers = ['zenn', 'qiita', 'note'];
  const tags = ['Python', 'TypeScript', 'Go', 'JavaScript'];

  const mockedTechArticles: typeof techArticles = {
    articles: [...Array(itemsCount)].map((_, i) => {
      const date = new Date(baseDate);
      date.setDate(date.getDate() - i * i);
      return {
        "title": `Article title ${i}`,
        "url": `https://example.com/articles/${i}`,
        "publishedAt": date.toISOString(),
        "likes": 123 * i,
        "publisher": publishers[i % publishers.length],
        "tags": tags.slice(
            i % tags.length,
            (i * i) % tags.length,
        )
      };
    }),
  };
  return mockedTechArticles;
});

describe('TechArticles', () => {
  describe('スナップショットテスト', () => {
    test('技術記事が描画される', () => {
      const { container } = render(<TechArticles />);
      expect(container).toMatchSnapshot();
    });
  });
});
