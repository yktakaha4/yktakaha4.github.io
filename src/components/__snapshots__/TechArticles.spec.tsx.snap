// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechArticles スナップショットテスト 技術記事が描画される 1`] = `
<div>
  <div
    style="margin-bottom: 0.2rem;"
  >
    <nav
      class="navbar"
      style="margin: 0px; padding: 0px; background-color: transparent; box-shadow: none; height: inherit;"
    >
      <div
        class="navbar__inner"
      >
        <div
          class="navbar__items"
        >
          <div
            class="navbar__search"
          >
            <input
              class="navbar__search-input"
              maxlength="100"
              placeholder="検索"
              type="text"
              value=""
            />
          </div>
          <div
            class="navbar__item"
          >
            77
             件
          </div>
        </div>
        <div
          class="navbar__items navbar__items--right"
        >
          <div
            class="navbar__item"
          >
            <ul
              class="pagination"
              style="margin: 0px;"
            >
              <li
                class="pagination__item"
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  «
                </a>
              </li>
              <li
                class="pagination__item pagination__item--active"
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  1
                </a>
              </li>
              <li
                class="pagination__item "
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  2
                </a>
              </li>
              <li
                class="pagination__item "
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  3
                </a>
              </li>
              <li
                class="pagination__item"
              >
                <span>
                  …
                </span>
              </li>
              <li
                class="pagination__item"
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  »
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  </div>
  <table
    style="margin: 0px;"
  >
    <thead>
      <tr>
        <th
          style="text-align: center; width: 10%;"
        >
          公開日
        </th>
        <th
          style="text-align: left; width: 70%;"
        >
          タイトル
        </th>
        <th
          style="text-align: left; width: 20%;"
        >
          タグ
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          2022/3/28
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/technology_for_sending_email"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/technology_for_sending_email"
          >
            電子メール送信に関する技術
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              221
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2022/5/9
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/documentation_inspired_by_http_cache"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/documentation_inspired_by_http_cache"
          >
            HTTPキャッシュに学ぶ、無理のないドキュメント更新運用
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              220
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2022/10/30
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/study_csrf_attack"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/study_csrf_attack"
          >
            CSRF(Cross-Site Request Forgery)攻撃について
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              189
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/4/25
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/running_an_agile_sre_team"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/running_an_agile_sre_team"
          >
            アジャイルなSREチームの運用
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              63
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2021/2/28
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/update_lp_for_maintenanceability"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/update_lp_for_maintenanceability"
          >
            HTML+jQueryでできたLPをwebpack+TypeScript+Cypressで楽しく保守する
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              51
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2021/1/27
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://note.com/yktakaha4/n/n67789c25ad73"
            rel="noopener noreferrer"
            target="_blank"
            to="https://note.com/yktakaha4/n/n67789c25ad73"
          >
            LAPRASの採用選考を受けて感じたCX(Candidate Experience)と、その重要性
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              note
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              49
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/7/1
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/history_books_on_information_technology"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/history_books_on_information_technology"
          >
            情報技術に関する歴史や文化の本
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              47
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2022/1/9
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/kintone-form-model-generator"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/kintone-form-model-generator"
          >
            TypeScript Compiler APIでKintoneの型情報を自動生成するCLIツールを作った
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              45
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2022/12/23
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/introducing_kunitori"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/introducing_kunitori"
          >
            リポジトリへの貢献量に応じて日本を手中に収めるCLIを作った
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              44
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2022/2/28
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://zenn.dev/yktakaha4/articles/production_freeze_manual"
            rel="noopener noreferrer"
            target="_blank"
            to="https://zenn.dev/yktakaha4/articles/production_freeze_manual"
          >
            エラーバジェット枯渇時のプロダクションフリーズ運用マニュアル
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <div
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 96c0-35.3 28.7-64 64-64H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H80c-44.2 0-80-35.8-80-80V128c0-17.7 14.3-32 32-32s32 14.3 32 32V400c0 8.8 7.2 16 16 16s16-7.2 16-16V96zm64 24v80c0 13.3 10.7 24 24 24H296c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24H184c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16h48c8.8 0 16-7.2 16-16s-7.2-16-16-16H384c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16H432c8.8 0 16-7.2 16-16s-7.2-16-16-16H176c-8.8 0-16 7.2-16 16z"
                />
              </svg>
              Zenn
            </div>
            <div
              class="badge badge--danger"
              style="white-space: nowrap;"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 0.2rem; vertical-align: bottom;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                />
              </svg>
              31
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
